libhb/work.c

```diff
@@ -1703,6 +1703,12 @@ static void do_job(hb_job_t *job)
         init.color_range = job->passthru_dynamic_hdr_metadata & DOVI &&
                             job->dovi.dv_profile == 5 ?
                             title->color_range : AVCOL_RANGE_MPEG;
+#if HB_PROJECT_FEATURE_QSV
+        if (hb_qsv_full_path_is_enabled(job))
+        {
+            init.color_range = (job->qsv.ctx->out_range == AVCOL_RANGE_UNSPECIFIED) ? title->color_range : job->qsv.ctx->out_range;
```

Comment:
For now in case if the input is full range and sw pipeline is limited, don't we need to do a explicit conversion from full -> limited range all the time? I don't see that sw filter inserted for this automatically and looks that full range input processed as limited. For QSV I just wanted to allow use input true color range as is if we don't use sw filters anyway.

Reply:
The conversion is done in decavcodec.c right now for the sw path, together with size and rotation.

Reply:
QSV full path doesn't do any conversion at the moment, it's related to issue #5179.
Anyway, no harm in keeping those few lines for now I guess.

Reply:
> The conversion is done in decavcodec.c right now for the sw path, together with size and rotation.
Why is conversion needed for e.g. 10b full range sw decoder -> (no filters) -> 10b full range sw encoder pipelines?

We are looking at #5179 separately.

Reply:
It's not, decavcodec.c converts only If needed, it's just that's HandBrake always set the job range to limited (expect for Dolby Vision 5), and there is no UI to set the preferred range apart from your specific QSV hack above.